overlay/network.hpp

```diff
@@ -102,50 +113,86 @@ class IPWrapper
     return *this;
   } 
 
-  net::IP address() const { return address_; }
-
 private:
-  net::IP address_;
   uint8_t index_;
 };
 
-inline std::ostream& operator<<(std::ostream& stream, const IPWrapper& _ip)
+inline Try<IP> IP::convert(const net::IP& ip)
+{
+  switch(ip.family()) {
+    case AF_INET:
+      return IP(ip.in().get());
+    case AF_INET6:
+      return IP(ip.in6().get());
+    default:
+      UNREACHABLE();
+  }
+}
+
+inline Try<IP> IP::parse(const std::string& value, int family)
 {
-  stream << _ip.address();
+  Try<net::IP> ip = net::IP::parse(value, family);
+  if (ip.isError()) {
+    return Error(ip.error());
+  }
+
+  switch(family) {
+    case AF_INET:
+      return IP(ip.get().in().get());
+    case AF_INET6:
+      return IP(ip.get().in6().get());
+    default:
+      UNREACHABLE();
+  }
+}
+  
+
+// Returns the string representation of the given IP using the
+// canonical form, for example: "10.0.0.1" or "fe80::1".
+inline std::ostream& operator<<(std::ostream& stream, const IP& ip)
+{
+  stream << ip;
   return stream;
 }
 
-class Network
+class Network : public net::IPNetwork
```

Comment:
After upgrading to the latest `libmesos` this will have to change to `net::IP::Network`.